apiserver/facades/agent/deployer/register.go

```diff
@@ -40,7 +40,7 @@ func NewDeployerFacade(ctx facade.ModelContext) (*DeployerAPI, error) {
 	}
 
 	controllerConfigGetter := ctx.ServiceFactory().ControllerConfig()
-	unitRemover := ctx.ServiceFactory().Unit()
+	unitRemover := ctx.ServiceFactory().Application(nil)
```

Comment:
I don't like the nil argument. 
There must be a better way to do that.

Reply:
This is still evolving; we are establishing patterns of use. It predates this PR and is used in several places. To remove a unit, no registry is required so there's no need to waste CPU and memory creating one. There's more refactoring to do. But not is this PR which is only dealing with moving existing code.

Reply:
I think this could be refactored further later, happy to approve for now but a TODO for refactoring would be good.

Reply:
Added TODO in next PR